.travis.yml

```diff
@@ -26,7 +26,7 @@ before_script:
 
   # Validate cow schema
   - composer global require silverstripe/cow ^2
-  - ~/.composer/vendor/bin/cow schema:validate
+  - if [[ -d ~/.config/composer ]]; then ~/.config/composer/vendor/bin/cow schema:validate; else ~/.composer/vendor/bin/cow schema:validate; fi
```

Comment:
```suggestion
  - ~/.config/composer/vendor/bin/cow schema:validate
```

Reply:
I think I'll use your suggestion on slack instead - the base I was trying to cover here is the odd event of somehow regressing in container assignment or something. It really never should, but who knows - you mentioned it happened "sometimes" on slack, thus my precaution.

It would also cover the `$COMPOSER_HOME` edge case (not that that is really relevant in a controlled environment), and is far less characters to type :>

```
$ composer global --help

[...]

Help:
  Use this command as a wrapper to run other Composer commands
  within the global context of COMPOSER_HOME.

  You can use this to install CLI utilities globally, all you need
  is to add the COMPOSER_HOME/vendor/bin dir to your PATH env var.

  COMPOSER_HOME is c:\Users<user>\AppData\Roaming\Composer on Windows
  and /home/<user>/.composer on unix systems.

  If your system uses freedesktop.org standards, then it will first check
  XDG_CONFIG_HOME or default to /home/<user>/.config/composer

  Note: This path may vary depending on customizations to bin-dir in
  composer.json or the environmental variable COMPOSER_BIN_DIR.
```

```diff
- if [[ -d ~/.config/composer ]]; then ~/.config/composer/vendor/bin/cow schema:validate; else ~/.composer/vendor/bin/cow schema:validate; fi
+ composer global exec -v -- cow schema:validate
```

Reply:
Steaming 😠 
Composer's `-d` flag is almost entirely pointless, it's effectively what `composer global` does, rather than what I expected - run a global command from the context of a directory. Instead it's more like `cd somewhere/random && ./lol`

Reply:
If it's too much work right now, just use the freedesktop.org standards suggestion (we've already done this in most of the other modules)

Reply:
Nah I just modified the order of operations and installed cow locally, after all the tests have run (`after_script`). Works well now. The final red blip is a `php 5.6` `segfault` issue on shutdown after running more than one test, where it includes [`testGetSummaryReturnsFileTitleWhenLinkedToFile`](https://github.com/silverstripe/silverstripe-elemental-fileblock/blob/560da9f2b037335846020549a70fdfa0bec9e091/tests/Block/FileBlockTest.php#L54)

Reply:
I can reset this to be a hard coded path to be consistent with other modules though if you'd prefer.

Reply:
Nah it's fine

Reply:
I've reproduced the segfault locally, will look at it now

Reply:
There are a few more details on slack, which I'll go copy paste now :)